src/test/java/com/puppycrawl/tools/checkstyle/utils/BlockCommentPositionTest.java

```diff
@@ -0,0 +1,114 @@
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.utils;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.internal.TestUtils;
+
+public class BlockCommentPositionTest {
+
+    @Test
+    public void testJavaDocsRecognition() throws Exception {
+        final List<BlockCommentPositionTestMetadata> metadataList = Arrays.asList(
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnClass.java",
+                        BlockCommentPosition::isOnClass, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnMethod.java",
+                        BlockCommentPosition::isOnMethod, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnField.java",
+                        BlockCommentPosition::isOnField, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnEnum.java",
+                        BlockCommentPosition::isOnEnum, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnConstructor.java",
+                        BlockCommentPosition::isOnConstructor, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnInterface.java",
+                        BlockCommentPosition::isOnInterface, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnAnnotation.java",
+                        BlockCommentPosition::isOnAnnotationDef, 3),
+                new BlockCommentPositionTestMetadata("InputBlockCommentPositionOnEnumMember.java",
+                        BlockCommentPosition::isOnEnumConstant, 2)
+        );
+
+        for (BlockCommentPositionTestMetadata metadata : metadataList) {
+            final DetailAST ast = TestUtils.parseFile(
+                    new File(getPath(metadata.getFileName()))
+            );
+            final int matches = foundJavaDocsCount(ast, metadata.getAssertion());
+            assertEquals(metadata.getMatchesNum(), matches);
+        }
+    }
+
+    private static int foundJavaDocsCount(DetailAST detailAST,
+                                          Function<DetailAST, Boolean> assertion) {
+        int matchFound = 0;
+        DetailAST node = detailAST;
+        while (node != null) {
+            if (node.getType() == TokenTypes.BLOCK_COMMENT_BEGIN
+                    && JavadocUtils.isJavadocComment(node)) {
+                assertTrue(assertion.apply(node));
```

Comment:
it is bad parctice to put "assertXXXX" in non "testXXXXXX" methods.
in your case -  just inline method "foundJavaDocsCount" as you used it ones.
If you see that it should be used - instead "assertXXX" do throw exception to fail a test for sure.